independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/PublisherResponseHandler.java

```diff
@@ -46,32 +47,98 @@ public void setStreamingResponseCustomizers(List<StreamingResponseCustomizer> st
         this.streamingResponseCustomizers = streamingResponseCustomizers;
     }
 
+    @SuppressWarnings("rawtypes")
     private static class SseMultiSubscriber extends AbstractMultiSubscriber {
 
+        private final Publisher publisher;
+        // 0: no items have been pushed by the stream
+        // 1: the first item has been pushed by the stream, and we have yet to send the empty buffer (with the headers)
+        // 2: the empty buffer (with the headers) was sent, and we have received a response
+        private volatile int state = 0;
+
         SseMultiSubscriber(ResteasyReactiveRequestContext requestContext, List<StreamingResponseCustomizer> staticCustomizers,
-                long demand) {
+                Publisher publisher, long demand) {
             super(requestContext, staticCustomizers, demand);
+            this.publisher = publisher;
+        }
+
+        @Override
+        public void onSubscribe(Subscription s) {
+            this.subscription = s;
```

Comment:
You need to ensure that you do not receive a subscription twice. In this case, you need to cancel the new one. Obviously, it means that the subscription must be stored in an atomic reference or in a volatile with the atomic companion object.

Reply:
It depends, you might want your operator to support fan-out, but it's not what we usually do indeed, so storing the subscription in a `AtomicReference<Flow.Subscription>` and cancelling when already subscribed is a common idiom.

Reply:
This can't happen in this case as we explicitly control when the subscription is made.